scene/animation/animation_player.cpp

```diff
@@ -1203,11 +1203,14 @@ void AnimationPlayer::play(const StringName &p_name, float p_custom_blend, float
 	}
 
 	_stop_playing_caches();
-
+	
+	if (p_from_end) {
+		c.current.pos = c.current.from->animation->get_length();
```

Comment:
That's wrong, it has to stay where it was, here you're accessing `c.current.from` before it has been modified.

Reply:
Please squash commits together when fixing it :)

Reply:
I think now it is right. First-time _pull requesting_.